test/client/enketo-webform.spec.js

```diff
@@ -0,0 +1,755 @@
+import i18next from 'i18next';
+import lodash from 'lodash';
+import applicationCache from '../../public/js/src/module/application-cache';
+import connection from '../../public/js/src/module/connection';
+import controller from '../../public/js/src/module/controller-webform';
+import event from '../../public/js/src/module/event';
+import formCache from '../../public/js/src/module/form-cache';
+import gui from '../../public/js/src/module/gui';
+import settings from '../../public/js/src/module/settings';
+import store from '../../public/js/src/module/store';
+
+/**
+ * @typedef {import('sinon').SinonStub<Args, Return>} Stub
+ * @template Args
+ * @template Return
+ */
+
+/**
+ * @typedef {import('sinon').SinonSandbox} Sandbox
+ */
+
+/**
+ * @typedef ServiceWorkerControllerStub
+ * @property {string} [scriptURL]
+ */
+
+/**
+ * @typedef ServiceWorkerRegistrationStub
+ * @extends {EventListener}
+ * @property {ServiceWorker<'activated' | 'activating'> | null} active
+ * @property {ServiceWorker<'installing'> | null} installing
+ * @property {string} scope
+ * @property {Stub<[], Promise<void>} update
+ */
+
+/**
+ * @typedef ServiceWorkerStub
+ * @template State
+ * @property {ServiceWorkerControllerStub} [controller]
+ * @property {Stub<[ string ], Promise<ServiceWorkerRegistrationStub>>} register
+ * @property {State} state
+ */
```

Comment:
At time of writing: some of these types were mistakenly left in from an earlier approach, but may still be relevant for tests that are as yet unwritten. Just want to note that this is likely noise at the moment.